libs/shared/ui-components/src/SocialCard/SocialCard.tsx

```diff
@@ -18,7 +18,16 @@ export const SocialCard: FC<TSocialCardProps> = ({
 }) => {
   const { asPath } = useRouter();
 
-  const url = `${process.env['NEXT_PUBLIC_VERCEL_URL']}${asPath}`;
```

Comment:
Instead of replacing the fallback machinery entirely, would it work to use `process.env['NEXT_PUBLIC_VERCEL_URL']` instead of `window.location.origin`?

That's more what I was expecting -- pulling Next's URL instead of the window location.

What you had before might have been close to working -- it looks like it was just missing the leading `https://`.

Reply:
But I think process.env['NEXT_PUBLIC_VERCEL_URL'] is already contains` https://`

Reply:
Hm. Do you know, @gabalafou?

When we saw [this spurious injection of the domain into the URL](https://github.com/Quansight/Quansight-website/pull/733#issuecomment-1570775167) before, it was because the `https://` scheme was missing from the front of `url`.

Reply:
@bskinn seems like adding` https://` works

Reply:
Per the docs, [VERCEL_URL does not include https://](https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables), so neither does `NEXT_PUBLIC_VERCEL_URL`. 

Sorry I missed that in my suggestion diff.